src/main/java/HW16_Stream_API/Product.java

```diff
@@ -0,0 +1,95 @@
+package HW16_Stream_API;
+
+import java.time.LocalDate;
+import java.util.*;
+import java.util.stream.Collectors;
+
+public class Product {
+    private String category;
+    private double price;
+    private boolean isWithDiscount;
+    private LocalDate date;
+
+    public Product(){}
+    public Product(String category, double price, boolean isWithDiscount) {
+        this.category = category;
+        this.price = price;
+        this.isWithDiscount = isWithDiscount;
+        this.date = LocalDate.now();
+    }
+    private final List<Product> productList = new ArrayList<>();
+
+    public void addProduct(Product product) {
+        productList.add(product);
+    }
+    public List<Product> getProductList(){
+        return productList.stream()
+                .filter(product -> product.price >= 250)
+                .filter(product -> product.category.equalsIgnoreCase("Book"))
+                .toList();
+    }
+
+    public List<Product> getDiscountedBooks(){
+        return productList.stream()
+                .filter(product -> product.category.equalsIgnoreCase("Book"))
+                .map(product -> {
```

Comment:
```suggestion
                .map(product -> new Product(product.category, product.price * 0.9, true))
```
лямбду краще в один рядок писати (якщо не виходить, то винести в окремий метод)